cryptarchia/cryptarchia.py

```diff
@@ -244,14 +244,11 @@ def tip(self) -> BlockHeader:
     def length(self) -> int:
         return len(self.blocks)
 
-    def contains_block(self, block: BlockHeader) -> bool:
-        return block in self.blocks
-
-    def block_position(self, block: BlockHeader) -> int:
-        assert self.contains_block(block)
+    def block_position(self, block: Id) -> int:
         for i, b in enumerate(self.blocks):
-            if b == block:
+            if b.id() == block:
                 return i
+        return -1
```

Comment:
nitpick: I would rather return an `Optional[int]` instead for the `-1` in case is not found.

Reply:
yeah, thank you.0722830bfccaea3319504770bd9d1aceb56e2e63

Reply:
Yeah. Thank you f35e7ba